tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/interceptor/ftPriorityChange/FaultToleranceInterceptorPriorityChangeAnnotationConfTest.java

```diff
@@ -0,0 +1,104 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.fault.tolerance.tck.interceptor.ftPriorityChange;
+
+import org.eclipse.microprofile.fault.tolerance.tck.interceptor.OrderQueueProducer;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+/**
+ * A Client to demonstrate the interceptors ordering behavior of FT and CDI annotations
+ * @author carlosdlr
+ */
+public class FaultToleranceInterceptorPriorityChangeAnnotationConfTest extends Arquillian {
+
+    @Inject
+    private InterceptorComponent testInterceptor;
+
+    @Inject
+    private OrderQueueProducer orderFactory;
+
+    @Deployment
+    public static WebArchive deploy() {
+        JavaArchive testJar = ShrinkWrap
+            .create(JavaArchive.class, "interceptorChangeFtAnnotation.jar")
+            .addClasses(InterceptorComponent.class, EarlyFtInterceptor.class, LateFtInterceptor.class, OrderQueueProducer.class)
+            .addAsManifestResource(new StringAsset("mp.fault.tolerance.interceptor.priority=3700"), "microprofile-config.properties")
+            .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
+            .as(JavaArchive.class);
+
+        return ShrinkWrap.create(WebArchive.class, "interceptorChangeFtAnnotation.war")
+            .addAsLibrary(testJar);
+    }
+
+    /**
+     * This test validates the interceptors execution order after call a method
+     * annotated with Asynchronous FT annotation, using a queue type FIFO (first-in-first-out).
+     * The head of the queue is that element that has been on the queue the longest time.
+     * In this case is validating that the early interceptor is executed at first.
+     * @throws InterruptedException if the current thread was interrupted while waiting
+     * @throws ExecutionException if the computation threw an exception
+     */
+    @Test
+    public void testAsync() throws InterruptedException, ExecutionException {
+        Future<String> result = testInterceptor.asyncGetString();
+        assertEquals(result.get(), "OK");
+        String [] expectedOrder = {"asyncGetString","EarlyOrderFtInterceptor","LateOrderFtInterceptor"};
```

Comment:
Similarly, should this not be `{"EarlyOrderFtInterceptor","LateOrderFtInterceptor", "asyncGetString"}` ?

Reply:
Hello @Azquelt,

This is not the case because the idea of this the is to validate that if the FT priority is changed to be called at first must work.